### PR TITLE
MXM OFI provider

### PR DESCRIPTION
**Pull request by shefty**
- EP_RDM, FI_TAGGED
    - Basic CQ, AV interfaces

Signed-Off-By: Valentin Petrov <valentin.petrov@intel.com>

### CODE REVIEW

**Comment by shefty** [comment]
I cannot test the MXM build, let alone execution.  This PR should be the same as what @vspetrov submitted, except for changes to the Makefile.am to handle a conflict with the merging of the psm2 provider.

Valentin, please confirm this is ready for merging.

File: src/fabric.c [hunk 1]
```diff
@@ -394,7 +394,10 @@ void fi_ini(void)
 	fi_register_provider(VERBS_INIT, NULL);
         /* Initialize the sockets provider last.  This will result in
            it being the least preferred provider. */
-	fi_register_provider(SOCKETS_INIT, NULL);
+        fi_register_provider(SOCKETS_INIT, NULL);
+#ifdef HAVE_MXM
+        fi_register_provider(MXM_INIT, NULL);
+#endif
```
Comment (shefty) [inline src/fabric.c]: This still needs to be fixed.  The sockets provider needs to be last, and the ifdef should not be needed.

**Comment by shefty** [comment]
Updated fabric.c

**Comment by vspetrov** [comment]
Hi, i'm back from vacation, hence a slow reply. I've tested this PR. It builds and runs ok. BTW, why you are saying you cannot build the provider. It is easy with to do with: --enable-mxm --with-mxm-lib=/nfs/inn/proj/mpi/users/vapetrov/mlnx_hpcx/mxm/lib/ --with-mxm-include=/nfs/inn/proj/mpi/users/vapetrov/mlnx_hpcx/mxm/include. All you need is to download Mellanox HPCX suite from their site.

**Comment by shefty** [comment]
> Hi, i'm back from vacation, hence a slow reply. I've tested this PR. It
> builds and runs ok. BTW, why you are saying you cannot build the provider.
> It is easy with to do with: --enable-mxm --with-mxm-
> lib=/nfs/inn/proj/mpi/users/vapetrov/mlnx_hpcx/mxm/lib/ --with-mxm-
> include=/nfs/inn/proj/mpi/users/vapetrov/mlnx_hpcx/mxm/include. All you
> need is to download Mellanox HPCX suite from their site.

I'm saying I can't because I don't have MXM installed.  I was wanting to know what I need to do for that.  Does the MXM software replace my kernel drivers?  (If so, that's a poor option for me.)  Does it work with the upstream libibverbs and libmlx4?

**Comment by vspetrov** [comment]
It's a user space library that can be installed anywhere in the system.
Yes, it works with ofed libibverbs and mlx4.

--
BR,
Valentin Petrov

sent from phone

**Pull request merged**